File: crm/types/calendarPeriod.ts
export const CALENDAR_PERIOD_STATUSES = ["em_andamento", "finalizado", "reprovado"] as const;
export type CalendarPeriodStatus = (typeof CALENDAR_PERIOD_STATUSES)[number];

export const CALENDAR_PERIOD_STATUS_LABELS: Record<CalendarPeriodStatus, string> = {
  em_andamento: "Em andamento",
  finalizado: "Finalizado",
  reprovado: "Reprovado",
};

export const CALENDAR_PERIOD_STATUS_COLORS: Record<CalendarPeriodStatus, string> = {
  em_andamento: "bg-white/10 text-white/70",
  finalizado: "bg-cyan/15 text-cyan",
  reprovado: "bg-red-400/15 text-red-400",
};

export type CalendarPeriod = {
  client_id: number;
  year: number;
  month: number;
  status: CalendarPeriodStatus;
};

File: backend/app/schemas/calendar_period.py
# backend/app/schemas/calendar_period.py
from pydantic import BaseModel, field_validator

CALENDAR_PERIOD_STATUSES = ["em_andamento", "finalizado", "reprovado"]


class CalendarPeriodOut(BaseModel):
    client_id: int
    year: int
    month: int
    status: str

    model_config = {"from_attributes": True}


class CalendarPeriodSetStatus(BaseModel):
    client_id: int
    year: int
    month: int
    status: str

    @field_validator("status")
    @classmethod
    def validate_status(cls, value: str) -> str:
        if value not in CALENDAR_PERIOD_STATUSES:
            raise ValueError(f"Status inválido. Use um de: {', '.join(CALENDAR_PERIOD_STATUSES)}")
        return value

    @field_validator("month")
    @classmethod
    def validate_month(cls, value: int) -> int:
        if not 1 <= value <= 12:
            raise ValueError("Mês inválido")
        return value
